src/communication/communication.service.ts

```diff
@@ -21,6 +21,15 @@ export class CommunicationService {
     this.loggedInUsers.delete(socket.did);
   }
 
+  /**
+   * Get the socket ID of a logged-in user by their DID
+   * @param did the user DID
+   * @returns socket ID if user is logged in, undefined otherwise
+   */
+  getLoggedInUser(did: string): string | undefined {
+    return this.loggedInUsers.get(did);
```

Comment:
perhaps this could throw an error if the did is not found... then could remove this logic from other places